storage/src/adb/keyless.rs

```diff
@@ -165,69 +161,165 @@ impl<E: Storage + Clock + Metrics, V: Codec, H: CHasher> Keyless<E, V, H> {
         )
         .await?;
 
-        // Rewind to the last commit point if necessary.
-        let mut op_index = locations_size;
-        // The most recent commit point (if any).
-        let mut last_commit_loc = None;
-        // The location of the first append operation to follow the last known commit, and the
-        // offset it wraps.
-        let mut rewind_point = None;
-        while op_index > 0 {
-            op_index -= 1;
-            let offset = locations.read(op_index).await?;
-            let section = op_index / cfg.log_items_per_section.get();
-            let op = log.get(section, offset).await?.expect("no operation found");
-            match op {
-                Operation::Commit(_) => {
-                    last_commit_loc = Some(op_index);
-                    break;
+        // Trim any locations/mmr elements that do not have corresponding operations in log.
+        let log_items_per_section = cfg.log_items_per_section.get();
+        let mut walked_back = 0;
+        let mut section_offset = None;
+        while aligned_size - walked_back > 0 {
+            let loc = aligned_size - walked_back - 1;
+            let offset = locations.read(loc).await?;
```

Comment:
Just asking the obvious question that this can't be past the tail index?

https://github.com/commonwarexyz/monorepo/blob/0a5c78c73871dd9971bd4cbae132a253775d36c9/storage/src/journal/fixed.rs#L423

Reply:
I suspect this is what `align_mmr_and_locations` does?